src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,117 @@
+import { useContext, useState } from 'react';
+import { TodoList } from '../TodoList.tsx/TodoList';
+import { TodosFilter } from '../TodosFilter/TodosFilter';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../TodoContext/TodoContext';
+import { ButtonFilter } from '../enum/ButtonFilter';
+
+export const TodoApp: React.FC = () => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const [title, setTitle] = useState('');
+  const [filterListTodos, setFilterListTodos] = useState('');
+
+  const onAdd = (value: Todo) => {
+    setTodos([...todos, value]);
+  };
+
+  const reset = () => {
+    setTitle('');
+  };
+
+  const onSubmit = (e:React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title.trim()) {
+      return;
+    }
+
+    onAdd({
+      title,
+      id: +new Date(),
+      completed: false,
+    });
+
+    reset();
+  };
+
+  const filterTodos = todos.filter((todo: Todo) => {
```

Comment:
wrap with useMemo those operations when you need to find or filter something in an array